src/util/simplify_expr2.cpp

```diff
@@ -415,13 +414,62 @@ static type2tc common_arith_op2_type(expr2tc &e, expr2tc &f)
 
 expr2tc add2t::do_simplify() const
 {
+  // (A + 1) + ~B --> A - B
+  if(is_add2t(side_1) && is_bitnot2t(side_2))
```

Comment:
You can transform this into a lambda function, then you will be able to handle the commutative operator

Reply:
in the lastest version this has alredy been modified  auto simplify_2 = [](const expr2tc &e, const expr2tc &f) -> expr2tc